spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/VersionedAppRegistryController.java

```diff
@@ -230,9 +239,41 @@ public void makeDefault(@PathVariable("type") ApplicationType type, @PathVariabl
 	@ResponseStatus(HttpStatus.OK)
 	public void unregister(@PathVariable("type") ApplicationType type, @PathVariable("name") String name,
 			@PathVariable("version") String version) {
+
+		AppRegistration defaultApp = appRegistryService.getDefaultApp(name, type);
+		if (defaultApp != null && defaultApp.getVersion().equals(version)) {
+			String streamWithApp = findStreamContainingAppOf(type, name);
+			if (streamWithApp != null) {
+				throw new AppUsedInStreamException(String.format("The app [%s:%s:%s] you're trying to unregister is " +
```

Comment:
This will result in a confusing error message as 

```
	public AppUsedInStreamException(String appId) {
		super(String.format("The '%s' application could not be found.", appId));
	}````